### DISCUSSION TITLE
Why scaladoc can not handle http doc hosting

### DISCUSSION THREAD

**Discussion by ghost** [scala/scala3#15561] [General Discussion]
By the document we can see:
```
cd scaladoc/output
python3 -m http.server 8080
```

How hard is it we have a builtin http static server hosting the document ?

why we need python?

We already have scala, who need a python3 as well?

suggest: why not put a http server inside scaladoc

``` bash

scaladoc serve -p :8080
```
that can hosting document in localhost:8080

**Comment by SethTisue** [comment]
I normally view generated Scaladoc without using any HTTP server at all — I just open the HTML files in my browser directly. But I see that the doc says:

> It’s not strictly necessary to go through an HTTP server, but because of CORS the documentation won’t work completely if you don’t

Note that JDK 17 added https://docs.oracle.com/en/java/javase/17/docs/api/jdk.httpserver/module-summary.html which could be used instead of Python for this — for users who are on JDK 17 or higher, anyway.

**Comment by julienrf** [comment]
FWIW, the IDE IntelliJ provides an HTTP server for you (click on a file, and then select Open in browser). This is what I use, currently.

I agree that it would be good to put something in the documentation that does not rely on Python, though… In which page of the documentation is this?

**Reply by ghost** [reply]
https://github.com/lampepfl/dotty/blob/main/scaladoc/README.md

I hope scaladoc itself can buitin a simple http server.

**Reply by julienrf** [reply]
(pinging @pikinier20)

**Comment by ghost** [comment]
```java
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import com.sun.net.httpserver.HttpServer;
```

I guess we can implement a very simple Http static file server only depends on JDK api

**Comment by ghost** [comment]
or any other better solution

**Comment by ckipp01** [comment]
Wanted to drop in here and say that this should 100% be a part of Scaladoc. I find it super odd that the Scaladoc instructions mention using Python to view them to get the full features. Just in general I don't think people will expect this, and apart from the readme, it wasn't part of the docs. I just opened [this](https://github.com/scala/docs.scala-lang/pull/2491) to try to make that clearer, but I really don't think this should be relying on Python.

**Reply by julienrf** [reply]
Note that there is also `sbt-site`, which does provide an HTTP server. Several documentation website generators (e.g. paradox) rely on `sbt-site` for watching the generated documentation. Maybe that solution should be considered? Integrating with `sbt-site` would also bring other benefits such as working out of the box with `sbt-ghpage`.

**Reply by ckipp01** [reply]
It'd be great if there could be a build-tool agnostic way.

**Comment by floating-cat** [comment]
JDK 18 also has provided a `jwebserver` command based on `com.sun.net.httpserver` which could use the below command to start an HTTP server: https://openjdk.org/jeps/408

```
jwebserver
// with specified port 
jwebserver -p 8080
```
